packages/connections/src/stores/connections-store.ts

```diff
@@ -363,6 +373,29 @@ export function useConnections(
           });
         }
       },
+      async removeConnection(connectionInfo: ConnectionInfo) {
+        await connectionStorage.delete(connectionInfo);
+        dispatch({
+          type: 'set-connections',
+          connections: connections.filter(
+            (conn) => conn.id !== connectionInfo.id
+          ),
+        });
+      },
+      async duplicateConnection(connectionInfo: ConnectionInfo) {
+        if (!connectionInfo.favorite) return;
```

Comment:
Is there ever a situation where a user can duplicate a non-favorite? Looks like you've already added a test that makes this action only available on favorites. Maybe we can remove this line and or have a separate FavoriteConnectionInfo type that has favorite as a required field?

Reply:
I think we can remove this line, because the menu does not even appear on non-favorite.
Whenever we decide to change that behavior, we wouldn't need to change this function then